conbench/entities/run.py

```diff
@@ -64,6 +65,7 @@ def create(data):
             )
             if github:
                 commit = Commit.create_github_context(sha, repository, github)
+                backfill_default_branch_commits(repository, commit)
```

Comment:
We should definitely confirm how long this takes and confirm what it would look like in realistic scenarios. 

I hear Elena's concerns, but personally I'm a bit less concerned because, if I am understanding correctly: 

* For apache/arrow we would almost never actually backfill, because each commit to dev is benchmarked already, and this only backfills if there are missing commits. And, in the (rare) cases when we do hit this, it should be at most a handful of commits not yet benchmarked. And, even then, it should only happen on the first benchmark posted, not on each benchmark after that, no?

* For repos that do merge commits, this could be a considerable number of commits (it will basically always be the number of commits that are on the branch... But even then, it'll be only on the first benchmark posted. But we definitely should investigate what that performance is like (though since we don't really yet support those, IMO if we deferred this investigation that would be OK, not ideal, but ok)

One other possible alternative would be to investigate if we could have this backfilling process be asynchronous, so that it doesn't interfere with the REST call lifecycle?

Finally, I suspect the label "backfill" here _might_ be a bit scary sounding — it's not intended to be like our typical backfill migrations, but more "fill in the gaps of history on the default branch"

Reply:
@ElenaHenderson Sorry for the confusion — this "backfill" (really it's more of a "check to see if we need to fill a small gap" like @jonkeane said) only happens when a new Run is created, *not* every time a new BenchmarkResult is created. 

In my understanding, for Arrow benchmarks, we only create a new Run on average every 30 minutes or so. I just timed this locally, and this code will increase the first `POST /benchmarks` time to about 1500 ms, but every subsequent post with the same `run_id` is completely unaffected, sitting around 20 ms. So that's only an increase of about a minute of build time per day.

Reply:
Actually, on top of that, this code is only run when a new Run is created on a commit that _doesn't already exist_ in the database. That only happens every 4 hours or so for Arrow.

Reply:
example: 
<img width="974" alt="image" src="https://user-images.githubusercontent.com/16600275/200899816-b6dbf510-6429-42f0-bd2b-98dfa81efb3a.png">

Reply:
@austin3dickey This is good. I completely missed the part that it will run only per RUN creation and not per BENCHMARK_RESULTS creation. I am not worried about this at all then.